src/react_native/utils.cljs

```diff
@@ -0,0 +1,35 @@
+(ns react-native.utils)
+
+(defonce ^:private throttle (atom {}))
+
+(defn- wrapped-ref
+  [{:keys [ref allow-multiple-presses?]} throttle-id]
+  (if allow-multiple-presses?
+    ref
+    (fn [ref-value]
+      (when ref-value
+        (when ref
+          (ref ref-value))
+        (js/setTimeout
+         #(let [id (hash ref-value)]
```

Comment:
why `ref-value` can't be used directly as id ?

Reply:
Hi, nice suggestion, using directly is working the same.
I was a little worried about the memory consumption of storing all ref objects and their state in global atoms across the app (as they are big JS objects).

Now I updated the code to just remove the entry from the map after timeout instead of changing the value to false.
(also updated the old `throttle-and-dispatch` function to do the same)

Reply:
Also, Thank you very much for your suggestion about the target received in the on-press event as an ID. But unfortunately, that didn't work due to an issue.

Issue: When tapping buttons quickly, sometimes the button re-renders itself or its parent, don't know exactly what happens but it causes that target-id to change. Change is very low, for example from 1030 to 1032, but it's hard to track.

So, I am keeping the current solution where storing ref-value as an id. 
This also gets affected due to the above issue. But in this case, it's easy to track because in this case id resets to original value (nil), and then we just ignore taps for these cases.